# Data Analysis 1
Data Set: Sales data from video games (https://www.kaggle.com/gregorut/videogamesales)  
Name: Toby  
Date: Dec.12.2018  




In [1]:
import pandas as pd
import numpy as np
import datetime as dt

In [2]:
# read csv files
vs_data_dir = r"/Users/toby/Downloads/vgsales.csv"
vs_data = pd.read_csv(vs_data_dir)
vs_data.head()

Rank                      Name Platform    Year         Genre Publisher  \
0     1                Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2         Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3            Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4         Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5  Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37

In [3]:
# vs_data.describe()

### Notice that the numbers quoted for sales are in the millions, and apply only for those games with over 100,000 sales

### 1. Which company is the most common video game publisher?

In [4]:
# can also do groupby().count()
vs_data['Publisher'].value_counts().head()

Electronic Arts                 1351
Activision                       975
Namco Bandai Games               932
Ubisoft                          921
Konami Digital Entertainment     832
Name: Publisher, dtype: int64

__ANS__: **EA** (Electronic Arts) is the most common video game publisher. They have published **1351** games in history.

### 2. What’s the most common platform?

In [5]:
vs_data['Platform'].value_counts().head(20)

DS      2163
PS2     2161
PS3     1329
Wii     1325
X360    1265
PSP     1213
PS      1196
PC       960
XB       824
GBA      822
GC       556
3DS      509
PSV      413
PS4      336
N64      319
SNES     239
XOne     213
SAT      173
WiiU     143
2600     133
Name: Platform, dtype: int64

__ANS__: **DS** turns out to be the most common platform! Yeah! They have holded more than **2163** games in history.

### 3. What about the most common genre?

In [6]:
vs_data['Genre'].value_counts().head(20)

Action          3316
Sports          2346
Misc            1739
Role-Playing    1488
Shooter         1310
Adventure       1286
Racing          1249
Platform         886
Simulation       867
Fighting         848
Strategy         681
Puzzle           582
Name: Genre, dtype: int64

__ANS__: **Action** is definitely the most common genre.

### 4. What are the top 20 highest grossing games?

In [7]:
print("Here's the top 20 highest grossing games in overall")
vs_data.sort_values('Global_Sales', ascending=False).head(20)

Here's the top 20 highest grossing games in overall


Rank                                          Name Platform    Year  \
0      1                                    Wii Sports      Wii  2006.0   
1      2                             Super Mario Bros.      NES  1985.0   
2      3                                Mario Kart Wii      Wii  2008.0   
3      4                             Wii Sports Resort      Wii  2009.0   
4      5                      Pokemon Red/Pokemon Blue       GB  1996.0   
5      6                                        Tetris       GB  1989.0   
6      7                         New Super Mario Bros.       DS  2006.0   
7      8                                      Wii Play      Wii  2006.0   
8      9                     New Super Mario Bros. Wii      Wii  2009.0   
9     10                                     Duck Hunt      NES  1984.0   
10    11                                    Nintendogs       DS  2005.0   
11    12                                 Mario Kart DS       DS  2005.0   
12    13                   Pokemon Gold/Pokemon Silver       GB  1999.0   
13    14                                       Wii Fit      Wii  2007.0   
14    15                                  Wii Fit Plus      Wii  2009.0   
15    16                            Kinect Adventures!     X360  2010.0   
16    17                            Grand Theft Auto V      PS3  2013.0   
17    18                 Grand Theft Auto: San Andreas      PS2  2004.0   
18    19                             Super Mario World     SNES  1990.0   
19    20  Brain Age: Train Your Brain in Minutes a Day       DS  2005.0   

           Genre               Publisher  NA_Sales  EU_Sales  JP_Sales  \
0         Sports                Nintendo     41.49     29.02      3.77   
1       Platform                Nintendo     29.08      3.58      6.81   
2         Racing                Nintendo     15.85     12.88      3.79   
3         Sports                Nintendo     15.75     11.01      3.28   
4   Role-Playing                Nintendo     11.27      8.89     10.22   
5         Puzzle                Nintendo     23.20      2.26      4.22   
6       Platform                Nintendo     11.38      9.23      6.50   
7           Misc                Nintendo     14.03      9.20      2.93   
8       Platform                Nintendo     14.59      7.06      4.70   
9        Shooter                Nintendo     26.93      0.63      0.28   
10    Simulation                Nintendo      9.07     11.00      1.93   
11        Racing                Nintendo      9.81      7.57      4.13   
12  Role-Playing                Nintendo      9.00      6.18      7.20   
13        Sports                Nintendo      8.94      8.03      3.60   
14        Sports                Nintendo      9.09      8.59      2.53   
15          Misc  Microsoft Game Studios     14.97      4.94      0.24   
16        Action    Take-Two Interactive      7.01      9.27      0.97   
17        Action    Take-Two Interactive      9.43      0.40      0.41   
18      Platform                Nintendo     12.78      3.75      3.54   
19          Misc                Nintendo      4.75      9.26      4.16   

    Other_Sales  Global_Sales  
0          8.46         82.74  
1          0.77         40.24  
2          3.31         35.82  
3          2.96         33.00  
4          1.00         31.37  
5          0.58         30.26  
6          2.90         30.01  
7          2.85         29.02  
8          2.26         28.62  
9          0.47         28.31  
10         2.75         24.76  
11         1.92         23.42  
12         0.71         23.10  
13         2.15         22.72  
14         1.79         22.00  
15         1.67         21.82  
16         4.14         21.40  
17        10.57         20.81  
18         0.55         20.61  
19         2.05         20.22

### 5. For North American video game sales, what’s the median?

In [8]:
NA_mean = vs_data["NA_Sales"].median()
print(f'The median of North American video game sales is { NA_mean }')

The median of North American video game sales is 0.08


### 6. For the top-selling game of all time, how many standard deviations above/below the mean are its sales for North America?

In [9]:
NA_sales_mean = vs_data['NA_Sales'].mean()
NA_sales_std = vs_data['NA_Sales'].std()
best_NA_sales = vs_data[vs_data['Name']=='Wii Sports']['NA_Sales'].values[0]
print(f'The top-selling game of all time is Wii Sports, \
\nit is { (best_NA_sales-NA_sales_mean)//NA_sales_std } standard \
deviation above the mean for North America sales data')

The top-selling game of all time is Wii Sports, 
it is 50.0 standard deviation above the mean for North America sales data


### 7. The Nintendo Wii seems to have outdone itself with games. How does its average number of sales compare with all of the other platforms?

In [10]:
print("Here's the average global sales data according to each platforms.")
print("We can see that, on average, the sales of games in Wii is ranked No.9")
tmp = vs_data.groupby('Platform').agg({'Global_Sales':'mean'})
tmp.sort_values('Global_Sales', ascending=False).reset_index().head(10)

Here's the average global sales data according to each platforms.
We can see that, on average, the sales of games in Wii is ranked No.9


Platform  Global_Sales
0       GB      2.606633
1      NES      2.561939
2      GEN      1.050370
3     SNES      0.837029
4      PS4      0.827679
5     X360      0.774672
6     2600      0.729925
7      PS3      0.720722
8      Wii      0.699404
9      N64      0.686144

### 8. What's the year with most published games, across all platforms?

In [11]:
# tmp = vs_data['Year'].value_counts().sort_index
print("As we can see, the first 10 years of 21st century is the booming time for video games!")
tmp = pd.DataFrame(vs_data['Year'].value_counts())
tmp.reset_index(inplace=True)
tmp.columns = ['Year','Sales']
tmp.sort_values('Sales', ascending=False)
# tmp.reset_index(inplace=False)
tmp.head(10)

As we can see, the first 10 years of 21st century is the booming time for video games!


Year  Sales
0  2009.0   1431
1  2008.0   1428
2  2010.0   1259
3  2007.0   1202
4  2011.0   1139
5  2006.0   1008
6  2005.0    941
7  2002.0    829
8  2003.0    775
9  2004.0    763

### 9. Culture difference. Are customers from NA, EU, and JP different from each other?

In [12]:
tmp = vs_data[['Genre',"NA_Sales","EU_Sales","JP_Sales"]].groupby("Genre").agg({
    "NA_Sales":"sum",
    "EU_Sales":"sum",
    "JP_Sales":"sum"
})
tmp

NA_Sales  EU_Sales  JP_Sales
Genre                                     
Action          877.83    525.00    159.95
Adventure       105.80     64.13     52.07
Fighting        223.59    101.32     87.35
Misc            410.24    215.98    107.76
Platform        447.05    201.63    130.77
Puzzle          123.78     50.78     57.31
Racing          359.42    238.39     56.69
Role-Playing    327.28    188.06    352.31
Shooter         582.60    313.27     38.28
Simulation      183.31    113.38     63.70
Sports          683.35    376.85    135.37
Strategy         68.70     45.34     49.46

In [13]:
# to compare it, we normalize each Sales data columnms, and then scale it

tmp["NA_Sales"] = (tmp["NA_Sales"]-tmp["NA_Sales"].mean()) / tmp["NA_Sales"].std()
tmp["EU_Sales"] = (tmp["EU_Sales"]-tmp["EU_Sales"].mean()) / tmp["EU_Sales"].std()
tmp["JP_Sales"] = (tmp["JP_Sales"]-tmp["JP_Sales"].mean()) / tmp["JP_Sales"].std()

tmp["NA_Sales"] = tmp["NA_Sales"] / tmp["NA_Sales"].max()
tmp["EU_Sales"] = tmp["EU_Sales"] / tmp["EU_Sales"].max()
tmp["JP_Sales"] = tmp["JP_Sales"] / tmp["JP_Sales"].max()

# alright..divide by max looks more intutitve here

print("Here we can see that NA players are very similar to EU players\n\
The only tiny two differences are \n1. NA players prefer platform games more than EU ones \n\
2. EU players like racing games more than NA ones.\n\n\
On the other hand, JP players are very different. They show nomral interest to the \n\
best sales genre in EU and NA (action). And heavily prefer Role-playing.")
tmp

Here we can see that NA players are very similar to EU players
The only tiny two differences are 
1. NA players prefer platform games more than EU ones 
2. EU players like racing games more than NA ones.

On the other hand, JP players are very different. They show nomral interest to the 
best sales genre in EU and NA (action). And heavily prefer Role-playing.


NA_Sales  EU_Sales  JP_Sales
Genre                                     
Action        1.000000  1.000000  0.213975
Adventure    -0.508605 -0.430581 -0.226846
Fighting     -0.278435 -0.315140 -0.082685
Misc          0.086294  0.040775  0.000715
Platform      0.158223 -0.003769  0.094739
Puzzle       -0.473471 -0.472021 -0.205435
Racing       -0.013013  0.110337 -0.207968
Role-Playing -0.075817 -0.045891  1.000000
Shooter       0.423098  0.342771 -0.283195
Simulation   -0.357145 -0.277705 -0.179324
Sports        0.619971  0.540129  0.113536
Strategy     -0.581101 -0.488907 -0.237511

### 10. Culture difference 2.0 Who are the most popular publisher at each market (NA, EU, JP)?

In [14]:
# tmp = vs_data[['Publisher',"NA_Sales","EU_Sales","JP_Sales"]].groupby("Publisher").agg({
#     "NA_Sales":"sum",
#     "EU_Sales":"sum",
#     "JP_Sales":"sum"
# })
# tmp.sort_values('NA_Sales', ascending=False).head()
# somehow this sort_value is not working...
tmp = vs_data[['Publisher',"NA_Sales"]].groupby("Publisher").agg({"NA_Sales":"sum",})
tmp.sort_values('NA_Sales', ascending=False).head()

NA_Sales
Publisher                            
Nintendo                       816.87
Electronic Arts                595.07
Activision                     429.70
Sony Computer Entertainment    265.22
Ubisoft                        253.43

In [15]:
tmp = vs_data[['Publisher',"EU_Sales"]].groupby("Publisher").agg({"EU_Sales":"sum",})
tmp.sort_values('EU_Sales', ascending=False).head()

EU_Sales
Publisher                            
Nintendo                       418.74
Electronic Arts                371.27
Activision                     215.53
Sony Computer Entertainment    187.72
Ubisoft                        163.32

In [16]:
tmp = vs_data[['Publisher',"JP_Sales"]].groupby("Publisher").agg({"JP_Sales":"sum",})
tmp.sort_values('JP_Sales', ascending=False).head()

JP_Sales
Publisher                             
Nintendo                        455.42
Namco Bandai Games              127.07
Konami Digital Entertainment     91.30
Sony Computer Entertainment      74.10
Capcom                           68.08

In [17]:
print("WOW..from the data above, you can see Nintendo is a pretty dominant publisher\n\
SONY has been a constant strong player across all regions. \n\
EU and NA are very similar in publisher ranking, and JP has their very own favorites.")

WOW..from the data above, you can see Nintendo is a pretty dominant publisher
SONY has been a constant strong player across all regions. 
EU and NA are very similar in publisher ranking, and JP has their very own favorites.
